trainlib/trainer.py

```diff
@@ -296,9 +296,17 @@ def _prepare_dirs(self) -> None:
         """Set up directories for saving logs, outputs and configs of current training session"""
         # create run_id, copy config file for reproducibility
         run_id = Path(self.config.run_id)
-        run_id.mkdir(exist_ok=True)
+        run_id.mkdir(exist_ok=True, parents=True)
         with open(run_id / "config.yaml", "w+") as f:
-            f.write(yaml.safe_dump(self.config))
+            config = dict(self.config)
+            # convert pathlib.Path to string, because of incompatibility with PyYAML
+            for path in ["run_id", "out_dir", "model_dir", "log_dir"]:
```

Comment:
Hardcoding things that are path: easy to  miss one. 

Instead could we check for `isinstance(config_value, Path)`?

Reply:
The problem is, that the config is nested. I'll have to write a function that goes recursively through all entries. 
There should be no possibility, that the user adds new paths.

I'll plan to rewrite the config parser to (A) get rid of munch and (B) allow #25 and (C) allow better hyperparameter tuning, directly from the config. I will then implement a better solution to check for paths.